examples/prefab/main.rs

```diff
@@ -0,0 +1,81 @@
+//! Demonstrates loading prefabs using the Amethyst engine.
+
+extern crate amethyst;
+extern crate rayon;
+
+use amethyst::assets::{PrefabLoader, PrefabLoaderSystem, RonFormat};
+use amethyst::config::Config;
+use amethyst::core::transform::{Transform, TransformBundle};
+use amethyst::input::{is_close_requested, is_key, InputBundle};
+use amethyst::renderer::{CameraPrefab, DisplayConfig, DrawShaded, Event, GraphicsPrefab, Light,
+                         ObjFormat, Pipeline, PosNormTex, RenderBundle, Stage, TextureFormat,
+                         VirtualKeyCode};
+use amethyst::{Application, Error, GameData, GameDataBuilder, State, StateData, Trans};
+
+type MyPrefabData = (
+    Option<GraphicsPrefab<ObjFormat, TextureFormat>>,
+    Option<Transform>,
+    Option<Light>,
+    Option<CameraPrefab>,
+);
+
+struct AssetsExample;
+
+impl<'a, 'b> State<GameData<'a, 'b>> for AssetsExample {
+    fn on_start(&mut self, data: StateData<GameData>) {
+        let StateData { world, .. } = data;
+        let prefab_handle = world.exec(|loader: PrefabLoader<MyPrefabData>| {
+            loader.load("prefab/example.ron", RonFormat, (), ())
+        });
+        world.create_entity().with(prefab_handle).build();
+    }
+
+    fn handle_event(&mut self, _: StateData<GameData>, event: Event) -> Trans<GameData<'a, 'b>> {
+        if is_key(&event, VirtualKeyCode::Escape) || is_close_requested(&event) {
+            Trans::Quit
+        } else {
+            Trans::None
+        }
+    }
+
+    fn update(&mut self, data: StateData<GameData>) -> Trans<GameData<'a, 'b>> {
+        data.data.update(&data.world);
+        Trans::None
+    }
+}
+
+fn main() {
+    if let Err(error) = run() {
+        eprintln!("Could not run the example!");
+        eprintln!("{}", error);
+        ::std::process::exit(1);
+    }
+}
+
+/// Wrapper around the main, so we can return errors easily.
+fn run() -> Result<(), Error> {
```

Comment:
This run function is no longer necessary, `main()` can return results.  This should mimic the other examples.